// @pxblue/highcharts/dist/utilities/i18n.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getChartsTooltipDateTimeLabelFormats = exports.getChartsAxisDateTimeLabelFormats = exports.getChartsLanguage = void 0;
var getChartsLanguage = function (i18nLanguage) {
    switch (i18nLanguage) {
        case 'fr':
            return {
                loading: 'Chargement...',
                months: [
                    'janvier',
                    'février',
                    'mars',
                    'avril',
                    'mai',
                    'juin',
                    'juillet',
                    'août',
                    'septembre',
                    'octobre',
                    'novembre',
                    'décembre',
                ],
                weekdays: ['dimanche', 'lundi', 'mardi', 'mercredi', 'jeudi', 'vendredi', 'samedi'],
                shortMonths: ['jan', 'fév', 'mar', 'avr', 'mai', 'juin', 'juil', 'aoû', 'sep', 'oct', 'nov', 'déc'],
                rangeSelectorFrom: 'Du',
                rangeSelectorTo: 'au',
                rangeSelectorZoom: 'Période',
                downloadPNG: 'Télécharger en PNG',
                downloadJPEG: 'Télécharger en JPEG',
                downloadPDF: 'Télécharger en PDF',
                downloadSVG: 'Télécharger en SVG',
                resetZoom: 'Réinitialiser le zoom',
                resetZoomTitle: 'Réinitialiser le zoom',
                thousandsSep: ' ',
                decimalPoint: ',',
            };
        default:
            return undefined;
    }
};
exports.getChartsLanguage = getChartsLanguage;
var getChartsAxisDateTimeLabelFormats = function (i18nLanguage) {
    switch (i18nLanguage) {
        case 'fr': // default to built-in locale until a native French speaker can specify
            return undefined;
        case 'en': // assume US English
            return {
                day: '%m/%d',
                hour: '%l:%M %P',
                millisecond: '%l:%M:%S %P',
                second: '%l:%M:%S %P',
                minute: '%l:%M',
                month: '%b, %Y',
                year: '%Y',
                week: '%b %e',
            };
        default:
            return undefined;
    }
};
exports.getChartsAxisDateTimeLabelFormats = getChartsAxisDateTimeLabelFormats;
var getChartsTooltipDateTimeLabelFormats = function (i18nLanguage) {
    switch (i18nLanguage) {
        case 'fr': // default to built-in locale until a native French speaker can specify
            return undefined;
        case 'en': {
            // assume US English
            var format = '%m/%d/%Y %l:%M:%S %P';
            return {
                day: format,
                hour: format,
                millisecond: format,
                second: format,
                minute: format,
                month: format,
                year: format,
                week: format,
            };
        }
        default:
            return undefined;
    }
};
exports.getChartsTooltipDateTimeLabelFormats = getChartsTooltipDateTimeLabelFormats;
